### PR TITLE
zigbee: Add device declaration for each sample

### PR DESCRIPTION
**Pull request by tomchy**
Since the HA declarations do not conform to the ZCL8 specification, each zigbee sample and application shall provide its own device definition.

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

### CODE REVIEW

**Comment by tomchy** [comment]
This is a draft PR. It cannot be merged as it refers to the ZCL8 APIs.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -73,9 +73,14 @@ Zigbee
     It is meant to be used only for debugging purposes.
   * Documentation page about :ref:`ug_zigbee_commissioning`.
   * Experimental support for Zigbee Green Power Combo Basic functionality.
+  * Zigbee device definition for each Zigbee sample and application.
+  * Identify handler in the :ref:`Zigbee Light Switch sample <zigbee_light_switch_sample>`.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:77]: I'd say this should go under Samples / Zigbee samples, further down, line 312ff.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -73,9 +73,14 @@ Zigbee
...
 
-* Updated :ref:`Zigbee shell library <lib_zigbee_shell>`.
-  For details, see `Libraries for Zigbee`_.
+* Updated:
+
+  * :ref:` Zigbee shell library <lib_zigbee_shell>`. For details, see `Libraries for Zigbee`_.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:81]:
```suggestion
  * :ref:` Zigbee shell library <lib_zigbee_shell>`.
    For details, see `Libraries for Zigbee`_.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -73,9 +73,14 @@ Zigbee
...
+  * Return code in the ZCL callbacks from the :c:macro:`RET_ERROR` to the :c:macro:`RET_NOT_IMPLEMENTED` if the device does not handle the ZCL command.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:82]:
```suggestion
  * Return code in the ZCL callbacks from :c:macro:`RET_ERROR` to :c:macro:`RET_NOT_IMPLEMENTED` if the device does not handle the ZCL command.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -73,9 +73,14 @@ Zigbee
...
+  * Enabled the ZCL8 compatibility mode: :c:macro:`ZB_ZCL_AUTO_MODE` inside the :c:func:`zigbee_default_signal_handler`.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:83]:
```suggestion
  * Enabled the ZCL8 compatibility mode :c:macro:`ZB_ZCL_AUTO_MODE` inside the :c:func:`zigbee_default_signal_handler`.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (tomchy) [reply doc/nrf/releases/release-notes-changelog.rst:77]: Moved!

**Review by greg-fer — APPROVED** [review]

**Review by sebastiandraus — APPROVED** [review]

File: west.yml [hunk 5]
```diff
@@ -109,7 +109,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: 10a433b07290b5df928a5471d681a9d41452c862
+      revision: pull/719/head
```
Comment (rlubos) [inline west.yml:112]: Please update SHA: `552bc9d1b5e8b604d19d5aa2f48843d9c32f0e18`

File: samples/zigbee/light_bulb/include/zb_dimmable_light.h [hunk 6]
```diff
@@ -0,0 +1,207 @@
+/*
+ * ZBOSS Zigbee 3.0
+ *
+ * Copyright (c) 2012-2020 DSR Corporation, Denver CO, USA.
+ * www.dsr-zboss.com
+ * www.dsr-corporation.com
+ * All rights reserved.
+ *
+ *
+ * Use in source and binary forms, redistribution in binary form only, with
```
Comment (rlubos) [inline samples/zigbee/light_bulb/include/zb_dimmable_light.h:10]: We should use regular Nordic license header here (`* SPDX-License-Identifier: LicenseRef-Nordic-5-Clause`) and in other files

**Review by rlubos — CHANGES_REQUESTED** [review]

File: west.yml (hunk 5 above)
Comment (tomchy) [reply west.yml]: updated!

File: samples/zigbee/light_bulb/include/zb_dimmable_light.h (hunk 6 above)
Comment (tomchy) [reply samples/zigbee/light_bulb/include/zb_dimmable_light.h]: Replaced!

**Review by rlubos — APPROVED** [review]

**Pull request merged**